File: 2202-maximize-the-topmost-element-after-k-moves/2202-maximize-the-topmost-element-after-k-moves.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maximumTop = function(nums, k) {
  if (!k) return nums[0];
  if (k == 1) return nums[1] ?? -1;
  if (nums.length === 1 && (k & 1) > 0) return -1;
  
  return Math.max(...nums.slice(0, k - 1), nums[k] || 0);
};

File: 2202-maximize-the-topmost-element-after-k-moves/2202-maximize-the-topmost-element-after-k-moves.py
class Solution:
    def maximumTop(self, nums: List[int], k: int) -> int:
        n = len(nums)
        if n == 1:
            return -1 if (k & 1) else nums[0]
        if k <= 1:
            return nums[k]
        if k > n:
            return max(nums)
        if k == n:
            return max(nums[:n-1])
        return max(max(nums[:k-1]), nums[k])